src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/BodyModelBinder.cs

```diff
@@ -132,7 +148,21 @@ public BodyModelBinder(IList<IInputFormatter> formatters, IHttpRequestStreamRead
                     return;
                 }
 
-                bindingContext.Result = ModelBindingResult.Success(model);
+                if (_isBindingRequired && model == null)
```

Comment:
I did consider triggering this behaviour based on line 108 of `InputFormatter`, but it didn't seem correct. There might be a nonempty request body that an `InputFormatter` subclass regards as "no input". The primary example would be a request whose body is just whitespace (or maybe even the string `null` - haven't tried that though) - in that case, `JsonInputFormatter` will behave identically to as if it were zero-length (and return `null`, which the developer is explicitly trying to test for when they check `ModelState.IsValid`).

Ideally, `InputFormatterResult` would have an `IsModelSet` property on it (equivalent to what we do for regular model binding). Then it could behave correctly for zero-length bodies, and other bodies that are regarded as "no input", whether the parameter is value-typed or not. However no such property exists, and adding one would be a breaking change if it defaults to `false` (because existing subclasses won't assign it). It seemed strange to add a new property there that defaults to `true`, but I guess we could do.

What do you think - should we add that property? Or do you have some other proposal about how we handle cases where the body content-length is nonzero but represents no value (according to the input formatter)?